### PR TITLE
Mysql HA scenario

### PR DESCRIPTION
**Pull request by Arau**
Not completely finished but it is good for a first review pass. 

There are a few details to polish (some lists are not rendering well yet).

### CODE REVIEW

**Comment by Arau** [comment]
https://katacoda.com/araustorageos

File: ha-mysql/assets/mysql-pod1.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+      - name: mysql
+        containerPort: 3306
+      volumeMounts:
+        - name: mysql-data
+          mountPath: /var/lib/mysql
+          subPath: mysql
+  volumes:
+    - name: mysql-data
+      persistentVolumeClaim:
+        claimName: mysqlpvc
```
Comment (aeroniero33) [inline ha-mysql/assets/mysql-pod1.yaml:24]: Add space here so it's easier to cat.

File: ha-mysql/assets/mysql-pod2.yaml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+      volumeMounts:
+        - name: mysql-data
+          mountPath: /var/lib/mysql
+          subPath: mysql
+  volumes:
+    - name: mysql-data
+      persistentVolumeClaim:
+        claimName: mysqlpvc
+  nodeSelector:
+    kubernetes.io/hostname: node02
```
Comment (aeroniero33) [inline ha-mysql/assets/mysql-pod2.yaml:26]: And space here.

File: ha-mysql/assets/storageos-pvc.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  name: mysqlpvc
+  labels:
+    storageos.com/replicas: "1"
+spec:
+  storageClassName: fast
+  accessModes:
+    - ReadWriteOnce
+  resources:
+    requests:
+      storage: 2Gi
```
Comment (aeroniero33) [inline ha-mysql/assets/storageos-pvc.yaml:13]: And space here.

File: ha-mysql/assets/storageos-service.yaml [hunk 4]
```diff
@@ -0,0 +1,15 @@
...
+spec:
+  ports:
+  - port: 5705
+    targetPort: 5705
+    name: http
+  type: LoadBalancer
+  selector:
+    name: storageos
+  externalIPs:
+  - 80.11.12.10
```
Comment (aeroniero33) [inline ha-mysql/assets/storageos-service.yaml:15]: And space here.

File: ha-mysql/assets/storageos-storageclass.yaml [hunk 5]
```diff
@@ -0,0 +1,11 @@
...
+apiVersion: storage.k8s.io/v1beta1
+metadata:
+  name: fast
+provisioner: kubernetes.io/storageos
+parameters:
+  pool: default
+  description: Kubernetes volume
+  fsType: ext4
+  adminSecretNamespace: default
+  adminSecretName: storageos-secret
```
Comment (aeroniero33) [inline ha-mysql/assets/storageos-storageclass.yaml:11]: And space here.

File: ha-mysql/040-move-pod.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+
+    `kubectl delete pod mysql`{{execute}}
+
+    > Wait until the Pod is deleted
+
+1. Recreate the Pod
+
+    `kubectl create -f mysql-pod1.yaml`{{execute}}
+
+1. Check the new Pod location
```
Comment (aeroniero33) [inline ha-mysql/040-move-pod.md:27]: Maybe we could explain with more detail that the pod is now deployed on a new node.

File: ha-mysql/050-stop-node.md [hunk 7]
```diff
@@ -0,0 +1,42 @@
...
+
+1. Check where the StorageOS replica is located
+
+    `kubectl exec -n kube-system cli -- storageos get volume -oyaml`{{execute}}
+
+    > Note the section `replicas.nodeName`
+
+1. SSH into the node
+
+    `ssh $node1`{{execute}}
```
Comment (aeroniero33) [inline ha-mysql/050-stop-node.md:35]: We should `-o StrictHostKeyChecking=no` to reduce friction.

File: ha-mysql/050-stop-node.md [hunk 8]
```diff
@@ -0,0 +1,42 @@
...
+    --command cli                            \
+    -- /bin/sh -c "while true; do sleep 999999; done"`{{execute}}
+
+1. The node $node1 holds the primary location
+
+    `echo $node1`{{execute}}
+
+1. Check where the StorageOS replica is located
+
+    `kubectl exec -n kube-system cli -- storageos get volume -oyaml`{{execute}}
```
Comment (aeroniero33) [inline ha-mysql/050-stop-node.md:29]: The cli pod was giving me a time out here, when using the --force pod delete.

I think not letting it terminate gracefully is not allowing to restart properly. Maybe we should remove the `--force --grace-period 0` flags.

File: ha-mysql/050-stop-node.md (hunk 7 above)
Comment (Arau) [reply ha-mysql/050-stop-node.md]: 100% thank you

File: ha-mysql/050-stop-node.md (hunk 8 above)
Comment (Arau) [reply ha-mysql/050-stop-node.md]: To avoid this behaviour I ended up removing the cli pod and installing as a binary.

**Pull request merged**